File: Code/CH9 Code/C#/WriteXML-c/Classify.cs
using System;
using System.IO;
using System.Text;
using System.Security.Cryptography;

namespace WriteXML_c
{
	/// <summary>
	/// Summary description for Classify.
	/// </summary>
  public class Classify
  {

    #region class local variables

    private static string EncryptKey = "CryptoManiac";
    private static MD5CryptoServiceProvider hashmd5;
    private static TripleDESCryptoServiceProvider des;
    private static byte[] pwdhash;

    #endregion

    static Classify()
    {
      hashmd5 = new MD5CryptoServiceProvider();
      pwdhash = hashmd5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(EncryptKey));

      //implement DES3 encryption
      des = new TripleDESCryptoServiceProvider();
      des.Key = pwdhash;
      des.Mode = CipherMode.ECB; //CBC, CFB
    }

    public static string Encrypt(string OriginalString)
    {
      byte[] buff = ASCIIEncoding.ASCII.GetBytes(OriginalString);
      return Convert.ToBase64String(des.CreateEncryptor().
                                        TransformFinalBlock(buff, 0, 
                                        buff.Length));
    }

    public static string Decrypt(string EncryptedString)
    {
      byte[] buff = Convert.FromBase64String(EncryptedString);
      return ASCIIEncoding.ASCII.GetString(des.CreateDecryptor().
                                               TransformFinalBlock(buff, 0, 
                                               buff.Length));
    }

  }
}

File: Code/CH9 Code/VB/WriteXML-vb/Classify.vb
Option Strict On

Imports System.IO
Imports System.Text
Imports System.Security.Cryptography

Public Class Classify

#Region "class local variables"

  Private Shared EncryptKey As String = "CryptoManiac"
  Private Shared hashmd5 As MD5CryptoServiceProvider
  Private Shared des As TripleDESCryptoServiceProvider
  Private Shared pwdhash() As Byte

#End Region

  Shared Sub New()
    hashmd5 = New MD5CryptoServiceProvider()
    pwdhash = hashmd5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(EncryptKey))

    'implement DES3 encryption
    des = New TripleDESCryptoServiceProvider()
    des.Key = pwdhash
    des.Mode = CipherMode.ECB
  End Sub

  Public Shared Function Encrypt(ByVal OriginalString As String) As String
    Dim buff() As Byte = ASCIIEncoding.ASCII.GetBytes(OriginalString)
    Return Convert.ToBase64String(des.CreateEncryptor(). _
                                      TransformFinalBlock(buff, 0, _
                                      buff.Length))
  End Function

  Public Shared Function Decrypt(ByVal EncryptedString As String) As String
    Dim buff() As Byte = Convert.FromBase64String(EncryptedString)
    Return ASCIIEncoding.ASCII.GetString(des.CreateDecryptor(). _
                                             TransformFinalBlock(buff, 0, _
                                             buff.Length))
  End Function

End Class
